idseq_pipeline/commands/accessionid2seq_functions.py

```diff
@@ -228,15 +248,29 @@ def get_sequence_by_accession_id_s3(accession_id, nt_loc_dict, nt_bucket, nt_key
     if entry:
         range_start = entry[0]+entry[1]
         seq_len = entry[2]
-        pipe_file = 'pipe-accession-%s' % accession_id
         accession_file = 'accession-%s' % accession_id
-        os.mkfifo(pipe_file)
-        get_range = "aws s3api get-object --range bytes=%d-%d --bucket %s --key %s %s" % (range_start, range_start + seq_len - 1, nt_bucket, nt_key, pipe_file)
-        get_range_proc = subprocess.Popen(get_range, shell=True, stdout=DEVNULL)
-        subprocess.check_call("tr -d '\n' < {pipe_file} > {accession_file}".format(pipe_file=pipe_file, accession_file=accession_file), shell=True)
-        exitcode = get_range_proc.wait()
-        assert exitcode == 0
-        os.remove(pipe_file)
+        NUM_RETRIES = 3
+        for attempt in range(NUM_RETRIES):
+            try:
+                pipe_file = 'pipe-{attempt}-accession-{accession_id}'.format(attempt=attempt, accession_id=accession_id)
+                os.mkfifo(pipe_file)
+                get_range = "aws s3api get-object --range bytes=%d-%d --bucket %s --key %s %s" % (range_start, range_start + seq_len - 1, nt_bucket, nt_key, pipe_file)
+                get_range_proc = subprocess.Popen(get_range, shell=True, stdout=DEVNULL)
+                subprocess.check_call("tr -d '\n' < {pipe_file} > {accession_file}".format(pipe_file=pipe_file, accession_file=accession_file), shell=True)
+                exitcode = get_range_proc.wait()
+                assert exitcode == 0
+                seq_len = os.stat(accession_file).st_size
```

Comment:
Oh I also added this line.   We now always get the newline-free character count.

Reply:
haha